.github/workflows/pytest.yml

```diff
@@ -66,6 +87,8 @@ jobs:
 
   ci-unit:
     runs-on: ubuntu-latest
+    needs: change_filter
+    if: ${{ needs.change_filter.outputs.run_code_checks == 'true' }}
```

Comment:
Maybe I'm not understanding but it seems like this will also mean that the unit tests won't run until we get into the merge queue, which may result in missing some basic failures until late in the PR process (syntax errors, imports, dependencies, etc). Is that really what we want?

Reply:
Duh. Yes, good catch. I added this in at the last second and totally forgot about why we even run unit tests in the first place. Taking this out.